phpseclib/Net/SSH2.php

```diff
@@ -2977,23 +2988,8 @@ public function ping(): bool
             return false;
         }
 
-        $this->window_size_server_to_client[self::CHANNEL_KEEP_ALIVE] = $this->window_size;
-        $packet_size = 0x4000;
-        $packet = Strings::packSSH2(
-            'CsN3',
-            MessageType::CHANNEL_OPEN,
-            'session',
-            self::CHANNEL_KEEP_ALIVE,
-            $this->window_size_server_to_client[self::CHANNEL_KEEP_ALIVE],
-            $packet_size
-        );
-
         try {
-            $this->send_binary_packet($packet);
-
-            $this->channel_status[self::CHANNEL_KEEP_ALIVE] = MessageType::CHANNEL_OPEN;
-
-            $response = $this->get_channel_packet(self::CHANNEL_KEEP_ALIVE);
+            $this->openChannel(self::CHANNEL_KEEP_ALIVE);
         } catch (\RuntimeException $e) {
             return $this->reconnect();
```

Comment:
The reconnect() is probably not desirable in the case that multiple channels have been disabled.

Perhaps more concerning is that with the current changes, ping() can be invoked at most 1 time when multiple channels are disabled, as subsequent pings will throw the multi channel exception.

Reply:
> Perhaps more concerning is that with the current changes, ping() can be invoked at most 1 time when multiple channels are disabled, as subsequent pings will throw the multi channel exception.

Decrementing `$this->channelCount` should resolve that.